lib/Basic/LangOptions.cpp

```diff
@@ -564,7 +565,9 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
   }
 
   // Set the "_pointerBitWidth" platform condition.
-  if (Target.isArch32Bit()) {
+  if (Target.isArch16Bit()) {
+    addPlatformConditionValue(PlatformConditionKind::PointerBitWidth, "_16");
```

Comment:
@kubamracek we will be able to update the test I added in #74818 to include this platform condition. But we'll need to get that PR merged first, as you surmised.